src/Form/Form.js

```diff
@@ -63,14 +75,16 @@ class Form extends Util.mixin(BindMixin) {
     let currentModel = this.buildStateObj(props.definition, 'value');
     let nextModel = this.buildStateObj(nextProps.definition, 'value');
     if (!Util.isEqual(currentModel, nextModel)) {
-      nextState.model = Util.extend({}, state.model, nextModel);
+      nextState.model = mergeNewModel(state.model, currentModel, nextModel);
```

Comment:
Let's test out this functionality, as it is going to impact all usages of `Form`